pkg/controller/v1beta1/trainedmodel/sharding/sharing_strategy.go

```diff
@@ -0,0 +1,20 @@
+package sharding
+
+import (
+	v1beta1api "github.com/kubeflow/kfserving/pkg/apis/serving/v1beta1"
+)
+
+type Strategy interface {
+	GetOrAssignShard(trainedModel *v1beta1api.TrainedModel) int
+}
+
+//TODO MemoryStrategy will be implemented in another PR
+type MemoryStrategy struct {
```

Comment:
Consider a separate file for this strategy

Reply:
Refactored